plugins/src/main/java/co/bound/codeartifact/CodeArtifactRepoProvider.java

```diff
@@ -0,0 +1,145 @@
+package co.bound.codeartifact;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import groovy.json.JsonOutput;
+import groovy.json.JsonSlurper;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import org.codehaus.groovy.runtime.DefaultGroovyMethods;
+import org.gradle.api.artifacts.repositories.MavenArtifactRepository;
+import org.gradle.api.provider.Property;
+import org.gradle.api.services.BuildService;
+import org.gradle.api.services.BuildServiceParameters;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.codeartifact.CodeartifactClient;
+import software.amazon.awssdk.services.codeartifact.CodeartifactClientBuilder;
+import software.amazon.awssdk.services.codeartifact.model.GetAuthorizationTokenRequest;
+import software.amazon.awssdk.services.codeartifact.model.GetAuthorizationTokenResponse;
+
+public abstract class CodeArtifactRepoProvider implements BuildService<CodeArtifactRepoProvider.Params> {
+
+  public void configureRepo(MavenArtifactRepository spec) {
+    Params params = getParameters();
+    String domain = getRequiredProperty(params.getDomain());
+    String accountId = getRequiredProperty(params.getAccountId());
+    String region = getRequiredProperty(params.getRegion());
+    String repo = getRequiredProperty(params.getRepo());
+
+    spec.setName("codeartifact");
+    configureCodeArtifactUrl(spec, domain, accountId, region, repo);
+    spec.credentials(pwd -> {
+      pwd.setUsername("aws");
+      if (!params.getOffline().get()) {
+        try {
+          pwd.setPassword(getToken(domain, accountId, region, params.getGradleUserHome().get().toPath()));
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+      }
+    });
+  }
+
+  private GetAuthorizationTokenResponse token = null;
+
+  public String getToken(String domain, String accountId, String region, Path gradleUserHome) throws IOException {
+    if (isValidToken(token)) {
+      return token.authorizationToken();
+    }
+
+    Path cacheFile = getCacheFile(domain, accountId, region, gradleUserHome);
+    if (Files.exists(cacheFile)) {
+      //noinspection unchecked
```

Comment:
This comment probably suppresses the warning in the IDE only.
The build log still warns:
```
> Task :plugins:compileJava
Note: /home/runner/work/codeartifact-gradle-plugin/codeartifact-gradle-plugin/plugins/src/main/java/co/bound/codeartifact/CodeArtifactRepoProvider.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
```

Putting `@SuppressWarnings("unchecked")` on the method would suppress the warning. And to narrow the scope, it may make sense to extract a private method for `(Map<String, Object>) new JsonSlurper().parse(cacheFile)` and suppress the warnings only there.

Reply:
Converted from json to use java Properties file instead